### ISSUE TITLE
RFC: `dusty reboot` to reboot daemon?

### ISSUE THREAD

**Issue by jsingle** [gamechanger/dusty#525]
Sometimes you want to cancel whatever the daemon is currently working on, and you need to run a couple launchctl commands to reset its state.  We could write a dusty command to make this easier

This would only work if the daemon is being run with launchctl.  It would also need to be run as `sudo dusty reboot`

**Comment by thieman** [comment]
If the thing we're actually trying to do here is force the daemon to stop whatever it's doing and go back to listening for commands, this could probably be better accomplished through signal handling. The permissions may be tricky, though, as we'd want the non-privileged user to be able to send a signal to the root-owned daemon.

I don't think it's a good idea to be putting any direct process management stuff into Dusty that we can avoid, as it'll make porting that functionality to Linux basically impossible. The SSH agent stuff is a necessary exception that is already bound to cause us some headaches when we try to port it.

**Comment by jsingle** [comment]
The client would still need to use psutil to find the right process (the daemon) to send the signal to, right?  Is that better than using launchctl?

**Comment by thieman** [comment]
The daemon could conceivably write a PID file correlated to the location of its socket, avoiding the need for process lookups. Still not a very clean solution.

Maybe the best thing here would be to just set up a separate socket for receiving interrupts?

**Comment by jsingle** [comment]
I like the PID file better than using a separate socket. If we have a separate socket we need multiple threads telling each other what to do

**Comment by thieman** [comment]
I think the socket is the right way to do it. Looking across processes for a matching name is super janky and breaks the implicit assumption that we have (via allowing Dusty socket to be configurable) that running multiple daemons should be doable.

**Comment by thieman** [comment]
Though really that assumption is already sort of violated because `/etc/dusty` is hardcoded.

**Comment by jsingle** [comment]
But if we write a PID file based on the socket name, we're not name-matching processes

**Comment by thieman** [comment]
True. I'm still not sure how the permissions would work out with the PID method, though.

**Comment by jsingle** [comment]
Yeah you might need to run `dusty reboot` as root
